test/run-tests.py

```diff
@@ -615,28 +675,66 @@ def main(args):
     out_dir = tempfile.mkdtemp(prefix='sexpr-wasm-')
     out_dir_is_temp = True
 
+  variables['out_dir'] = os.path.abspath(out_dir)
+
+  allProcs = []
+  continuedErrors = 0
   try:
-    for i, p in enumerate(range(num_proc)):
-      args = (i, options, variables, inq, outq, out_dir)
-      proc = multiprocessing.Process(target=ProcessWorker, args=args)
-      proc.start()
+    if not options.no_multithreaded:
+      inq = multiprocessing.Queue()
+      outq = multiprocessing.Queue()
+      for info in infosToRun:
+        inq.put_nowait(info)
+      runWorkers = multiprocessing.Value('i', 1)
```

Comment:
Hm, maybe this is just a common thing I've done then:
```
bool running = true;
while (running) {
  ...
  if (something) running = false;
  ...
}
```
Seems like the same idea.

I don't really like `do_work` because it sounds like a function (same issue I have with `runWorkers`). Maybe `should_continue` or `should_run`?